### PR TITLE
New freeze option to deactivate block, tx freeze / Fix block options passing

### LINKED ISSUE #940: Object.freeze(this) in a constructor restricts class extension
#923, #907 

Object.freeze(this) is called at the end of the Transaction, Block, and BlockHeader constructors. 

Surprisingly, if you extend any of these classes and add properties, the derived class's constructor throws. 

**Example**
```js
const Transaction = require('@ethereumjs/tx').Transaction

class SpecialTransaction extends Transaction {
  constructor(){
    super()
    this.special = true;
  }
}

new SpecialTransaction()
```

Throws
```
TypeError: Cannot add property special, object is not extensible
```

Am guessing the freeze locks the outputs of the new generators but I wonder if locating it in the constructor might be too restrictive for users who want to use these libs as a base for other work. 

At ganache-core, for example, [Transaction is extended here][1]. 

[1]: https://github.com/trufflesuite/ganache-core/blob/1177fe8a2711066a1b082b15d22585158001e80d/lib/utils/transaction.js#L141

### PR DESCRIPTION
**Pull request by holgerd77**
Fixes #940

### CODE REVIEW

**Comment by jochem-brouwer** [comment]
I was just working on some extra tests here to verify that these `opts` are passed and then realized that we also need this `freeze` in `BlockHeader`, will add here.

File: packages/block/src/types.ts [hunk 1]
```diff
@@ -43,6 +43,17 @@ export interface BlockOptions {
    * difficulty takes precedence over a provided static `difficulty` value.
    */
   calcDifficultyFromHeader?: BlockHeader
+  /**
+   * A block object by default gets frozen along initialization. This gives you
```
Comment (cgewecke) [inline packages/block/src/types.ts:47]:
```suggestion
A block object by default gets frozen during initialization. This gives you
```
Just a tiny idiomatic change here...

This comment block is really helpful...makes the rationale for freezing very clear.

**Review by cgewecke — APPROVED** [review]
Yes, this looks like a good solution to me! Tests are solid 💯

File: packages/block/src/block.ts [hunk 2]
```diff
@@ -70,7 +70,7 @@ export class Block {
       uncleHeaders.push(BlockHeader.fromValuesArray(uncleHeaderData, opts))
     }
 
-    return new Block(header, transactions, uncleHeaders)
+    return new Block(header, transactions, uncleHeaders, opts)
```
Comment (cgewecke) [inline packages/block/src/block.ts:73]: 👍

File: packages/block/src/types.ts (hunk 1 above)
Comment (holgerd77) [reply packages/block/src/types.ts:47]: Will leave this out since I don't want to dismiss the approval, hope that's ok.

**Pull request merged**